libraries/chain/combined_database.cpp

```diff
@@ -161,30 +161,30 @@ namespace eosio { namespace chain {
        : backing_store(backing_store_type::ROCKSDB), db(chain_db), kv_database{ [&]() {
             rocksdb::Options options;
 
-            options.create_if_missing = !cfg.read_only; // Creates a database if it is missing
+            options.create_if_missing = true; // Creates a database if it is missing
             options.level_compaction_dynamic_level_bytes = true;
-            options.bytes_per_sync = 1048576; // used to control the write rate of flushes and compactions.
+            options.bytes_per_sync = cfg.persistent_storage_bytes_per_sync; // used to control the write rate of flushes and compactions.
 
             // By default, RocksDB uses only one background thread
             // for flush and compaction.
             // Good value for `total_threads` is the number of cores
```

Comment:
@victorj8 - we should remove the 3 lines of comments and replace with: 
// Number of threads used for flush and compaction.

Reply:
Done